app/controllers/application_controller.rb

```diff
@@ -21,10 +21,7 @@ def with_current_request
   end
 
   def current_tenant(current_user)
-    tenant = current_user.tenant
-    found_tenant = Tenant.find_or_create_by(:external_tenant => tenant) if tenant.present?
-    return found_tenant if found_tenant
-    raise Catalog::NoTenantError
```

Comment:
I think this is the only place `Catalog::NoTenantError` is raised, so we can also remove the `rescue` above and the definition in `lib/exceptions.rb`.